### ISSUE TITLE
a11y: Desktop "Region" dropdown menu overall inaccessible

### ISSUE THREAD

**Issue by patrickhlauke** [bitwarden/clients#5527]
### Steps To Reproduce

1. Log out form all accounts
2. On the login screen, use the keyboard `Tab` to navigate
3. Try to reach the "Region" dropdown trigger and open the dropdown menu

### Expected Result

The dropdown trigger should receive focus, and the menu should then be openable using `Space`

### Actual Result

The dropdown trigger does not receive keyboard focus, making it impossible to open the dropdown using the keyboard.

### Screenshots or Videos

![screenshot of the login screen with the dropdown control highlighted](https://github.com/bitwarden/clients/assets/895831/d67e21ff-5687-49d3-bb31-6bcb9caed9a2)

![screenshot of the ticked/selected "US" option in the dropdown, and the NVDA output showing that the selection is not conveyed](https://github.com/bitwarden/clients/assets/895831/b5cb1c6f-bad0-4799-b9f9-169ecc7cdaa8)

### Additional Context

The dropdown control is currently implemented as an `<a>` element without an `href`. These anchors don't receive keyboard focus.

```html
<a cdkoverlayorigin="" aria-haspopup="menu" aria-controls="cdk-overlay-container" class="ng-tns-c13-1" ng-reflect-ng-switch="US">
    <label class="text-primary ng-tns-c13-1 ng-star-inserted">US</label><!--bindings={
  "ng-reflect-ng-switch-case": "US"
}-->
    <!--bindings={
  "ng-reflect-ng-switch-case": "EU"
}-->
    <!--bindings={
  "ng-reflect-ng-switch-case": "Self-hosted"
}-->
    <i aria-hidden="true" class="bwi bwi-fw bwi-sm bwi-angle-down ng-tns-c13-1"></i>
  </a>
```

At a minimum, the `<a>` needs an explicit `tabindex="0"` to be focusable. However, as this is a control that opens/closes a dropdown, rather than a link that navigates somewhere, it should really be implemented as a `<button>` element.

Note that once opened, focus is not moved to the menu/dropdown either. In all, this component is really badly and inaccessibly implemented.

Lastly, in the dropdown, the currently selected region button does not convey the fact that it's selected (while visually it has a tick-mark).

```
<button type="button" class="environment-selector-dialog-item ng-tns-c13-1">
    <i aria-hidden="true" class="bwi bwi-fw bwi-sm bwi-check ng-tns-c13-1" style="…"></i>
    <img alt="" class="img-us ng-tns-c13-1">
    <span class="ng-tns-c13-1">US</span>
</button>
```

At a minimum, this should have some form of programmatic indication that it's selected, such as `aria-pressed="true"` or similar

### Operating System

Windows

### Operating System Version

11

### Installation method

Direct Download (from bitwarden.com)

### Build Version

Version 2023.4.1 Shell 24.1.1 Renderer 112.0.5615.50 Node 18.14.0 Architecture x64

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by patrickhlauke** [comment]
#5532 makes a first attempt at addressing at least the keyboard access to the Region control itself, and conveys the state of the region button in the dropdown/modal using `aria-pressed`.

This still leaves the focus behaviour and modal nature (if you want to indeed make the dropdown a `role="modal"`) unresolved though. This will need further, more fundamental work.

**Comment by SergeantConfused** [comment]
Hi @patrickhlauke,

Thank you for this report. Just to make sure that you and I are on the same page, could you please confirm how you acquired version 2023.4.1?


Thank you in advance,

**Comment by patrickhlauke** [comment]
@SergeantConfused most recent master branch on the repo here

**Comment by SergeantConfused** [comment]
Hi @patrickhlauke,

Thank you, we are looking into this. If you have anything you'd like to add, please feel free to do so in this thread.


All the best,

**Comment by trmartin4** [comment]
Hi @patrickhlauke, are you OK closing this, as it was implemented with #5824?

**Comment by patrickhlauke** [comment]
@trmartin4 oops ... yes, forgot all about this one. as the control is now keyboard-accessible, and works well with screen readers (and hasn't regressed since, I just checked latest version), absolutely, this is closed by #5824
